### PR TITLE
[14.0] [FIX] fieldservice_sale_recurring: Only invoice invoiceable fsm order

### PR DESCRIPTION
**Pull request by paradoxxxzero**
Splitting of @mourad-ehm #944 PR
Depends on #1051, #1052

> Make link between invoice and fsm order for recurring order

### CODE REVIEW

**Review by mourad-ehm — APPROVED** [review]

**Review by hparfr — APPROVED** [review]

File: fieldservice_sale_recurring/__manifest__.py [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "depends": [
         "fieldservice_recurring",
         "fieldservice_sale",
+        "fieldservice_account",
```
Comment (max3903) [inline fieldservice_sale_recurring/__manifest__.py:14]: @paradoxxxzero This is useless: `fieldservice_sale` already depends on `fieldservice_account`.

**Review by max3903 — APPROVED** [review]
Minor change.

**Review by mourad-ehm — APPROVED** [review]

**Comment by ivantodorovich** [comment]
Hello!

It looks like this PR was merged, but not https://github.com/OCA/field-service/pull/1052, which is a direct dependency of this one.

As a result, we're calling a method `_get_invoiceable_fsm_order` (and overriding `_get_invoiceable_fsm_order_domain`), methods that don't exist in the `fieldservice_sale` module, as they were introduced by https://github.com/OCA/field-service/pull/1052

Strangely, we've just come across this issue in `17.0`, which means this bug has been migrated version after version unnoticed 😢 preventing a really basic workflow: invoicing.

```
AttributeError: 'sale.order.line' object has no attribute '_get_invoiceable_fsm_order'
```

I'm working on a fix at the moment

**Comment by ivantodorovich** [comment]
Reopened https://github.com/OCA/field-service/pull/1052 here:
- https://github.com/OCA/field-service/pull/1310

**Pull request merged**